libmamba/src/api/update.cpp

```diff
@@ -127,6 +127,24 @@ namespace mamba
         }
         else
         {
+            if (remove_not_specified)
+            {
+                auto hist_map = prefix_data.history().get_requested_specs_map();
+                std::vector<std::string> remove_specs;
+                for (auto& it : hist_map)
+                {
+                    if (std::find(update_specs.begin(), update_specs.end(), it.second.name)
+                        == update_specs.end())
+                    {
+                        remove_specs.push_back(it.second.name);
+                    }
+                }
+                solver.add_jobs(remove_specs, SOLVER_ERASE | SOLVER_CLEANDEPS);  // TODO it doesn't
```

Comment:
The wanted behavior here is to remove `remove_specs` **and** their dependencies, but only the specs are removed here.
Adding `SOLVER_CLEANDEPS` doesn't seem to be working... 
Any ideas @AntoinePrv ?

Reply:
Does it work with only `SOLVER_ERASE`?

Reply:
Yes, adding `SOLVER_CLEANDEPS` doesn't change anything on that matter (I just left it for now to illustrate).

Reply:
I think that it was just some mixing with the installation steps locally when I first tested it but is working fine now with the `micromamba` tests 👍